### PR TITLE
feat: course upload and student import.

### LINKED ISSUE #54: Upload Students Endpoint/API
As a Teacher, I want to be able to upload student details into the system via a CSV file and they become students in the system for my specific course.

**Required Parameter(s):**
- Course Id
- Uploaders Id

**CSV Field**
- Name
- Matric Number
- Level

Note: Matric Number will serve as their login ID, you can create a default password of 123456. Though we need to enforce a change of password on first login.

### PR DESCRIPTION
**Pull request by bynalab**

### CODE REVIEW

**Review by abdullahikawu — APPROVED** [review]
Looks good.

File: app/Http/Controllers/Api/CourseController.php [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        if ($validator->fails()) {
+            return response()->json(['error' => "All fields are required"], 400);
+        }
+
+        $id = Util::uuid();
+        $school_id = $request->get('school_id');
+        $faculty_id = $request->get('faculty_id');
+        $title = $request->get('title');
+        $code = $request->get('code');
+        $status = $request->get('status') ?? 'Active';
```
Comment (Auwalms) [inline app/Http/Controllers/Api/CourseController.php:38]: there is supposed to be an added field called `description`, you seems to have ommited it @bynalab. Can you confirm from the specs/task?

File: app/Http/Controllers/Api/UserController.php [hunk 2]
```diff
@@ -2,83 +2,70 @@
...
+            $checkDuplicateMatricNumber = User::where(['matric_number' => $customerArr[$i]['matric_number']])->first();
+            
+            if($checkDuplicateMatricNumber == null){
+                $data[] = [
+                    'id' => $this->uuid(),
+                    'matric_number' => $customerArr[$i]['matric_number'],
+                    'first_name' => $customerArr[$i]['first_name'],
+                    'other_names' => $customerArr[$i]['other_names'],
+                    'gender' => $customerArr[$i]['gender'],
+                    'password' => $customerArr[$i]['password'],
```
Comment (Auwalms) [inline app/Http/Controllers/Api/UserController.php]: @bynalab can we add default password say 123456? Also wondering if its possible to force them to change their password on their first login.

File: database/migrations/2021_02_14_150913_create_courses_table.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('courses', function (Blueprint $table) {
+            $table->string('id')->primary();
+            $table->string('school_id');
+            $table->string('faculty_id');
+            $table->string('title');
+            $table->string('code');
```
Comment (Auwalms) [inline database/migrations/2021_02_14_150913_create_courses_table.php:21]: Course `description` missing here. @bynalab

File: public/students/data.csv [hunk 4]
```diff
@@ -0,0 +1,4 @@
+first_name,other_names,matric_number,gender,password
+Habib,Endurance,U16/FNS/MTH/1012,male,123456
+Tajudeen, Jimoh,U16/FNS/MTH/1017,male,123456
```
Comment (Auwalms) [inline public/students/data.csv]: I think its best to programmatically add the password than to expect the Lecturers to add it the csv. Dem fit f-up @bynalab

**Review by Auwalms — CHANGES_REQUESTED** [review]
Please review the inline comments.

File: public/students/data.csv (hunk 4 above)
Comment (bynalab) [reply public/students/data.csv]: Noted, will do.

File: app/Http/Controllers/Api/UserController.php (hunk 2 above)
Comment (bynalab) [reply app/Http/Controllers/Api/UserController.php]: Yea, about the force change of password, i can find a way to notify the front-end about the redirection to change-password page or so, like a boolean or something, to say 'hey, this person needs to change his/her password' 
Think that will do, do you?

Comment (Auwalms) [reply app/Http/Controllers/Api/UserController.php]: Yes that works. Could be something like `defaultPassword: true|false`, in the case of true, they should be redirected back to the change password page.

**Review by Auwalms — APPROVED** [review]
Nice one

**Pull request merged**